### ISSUE TITLE
[a11y]: Overflow menu has 3 violations: The element with role "button" contains descendants with roles "xxx" which are ignored by browsers

### ISSUE THREAD

**Issue by cvikram21** [carbon-design-system/carbon#13088]
### Package

carbon-components, carbon-components-react

### Browser

Firefox

### Operating System

MacOS

### Package version

7.57.4

### React version

17.0.2

### Automated testing tool and ruleset

IBM Equal Access Accessibility Checker: Latest Deployment

### Description

The overflow menu when opened shows 3 violations in IBM Equal Access Accessibility Checker.
This can be reproducible in v10 and v11 storybooks

Screenshot from v10 storybook
<img width="1481" alt="image" src="https://user-images.githubusercontent.com/22382586/216676906-50c50fa0-5e35-4086-a148-b235cb434cfd.png">

### Reproduction/example

https://v10.carbondesignsystem.com/components/overflow-menu/usage/

### Steps to reproduce

1. Go to the v10 storybook for the overflow menu.
2. Open the menu.
3. Run the accessibility checker while keeping the menu open and the above-mentioned violations are seen.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by sstrubberg** [comment]
hey @cvikram21, thanks for submitting the issue. Can you re-run the Accessibility Checker against this instead? https://v7-react.carbondesignsystem.com/iframe.html?id=components-overflowmenu--default&args=&viewMode=story 

thx!

**Comment by QxiaoQ** [comment]
> hey @cvikram21, thanks for submitting the issue. Can you re-run the Accessibility Checker against this instead? https://v7-react.carbondesignsystem.com/iframe.html?id=components-overflowmenu--default&args=&viewMode=story
> 
> thx! Closing for now. Hit us up if you find anything!

Hi @sstrubberg , The link you provided here is the storybook of the **default overflowmenu** , but we are not talking about this, we are talking the overflowmenu with param **data-floating-menu-container=true**.
Although it can not reproduce the issue in this link, but I believe when you add the param **data-floating-menu-container=true** to this component in this version. And then scan by Accessibility Checker . This can be reproduced.

Unfortunately , there is not an enterpoint for me to do code changes in [https://v7-react.carbondesignsystem.com] to verify the accessibility by adding "data-floating-menu-container=true".

But this is indeed a problem, because  @cvikram21 mentioned that it can be reproduced in v10 and v11. Right?

**Comment by QxiaoQ** [comment]
@sstrubberg @cvikram21 Can anyone help reopen this issue?
